### PR TITLE
Refactor ajax and api services to not use jquery

### PR DESCRIPTION
**Pull request by sorta**
Split from https://github.com/travis-ci/travis-web/pull/2203, which ended up getting a bit large.

### CODE REVIEW

File: app/services/api.js [hunk 1]
```diff
@@ -30,49 +26,10 @@ export default Service.extend({
...
-          reject(jqXHR);
-        });
-      };
-
-      options.success = (payload, textStatus, jqXHR) => {
-        run(() => {
-          resolve(payload);
-        });
-      };
+    options.headers['Accept'] = 'application/json';
```
Comment (artursmirnov) [inline app/services/api.js]: Is this still required? It seems like `ajax` adds it by default, doesn't it? Though, i might be missing something 😅

File: app/routes/settings.js [hunk 2]
```diff
@@ -59,7 +59,7 @@ export default TravisRoute.extend({
     if (config.endpoints.sshKey) {
       const repo = this.modelFor('repo');
       const url = `/repos/${repo.get('id')}/key`;
-      return this.ajax.get(url, (data) => {
+      return this.ajax.get(url).then((data) => {
```
[the remark this answers is not in the document]
Comment (sorta) [reply app/routes/settings.js]: Thanks! Yes, this seems like a good idea to me, and seems to work. Change made! :)

File: app/services/api.js (hunk 1 above)
Comment (sorta) [reply app/services/api.js]: Good point, I think you're right. There were so many things like this where the api and ajax service were doing the same thing, but sometimes in slightly different ways haha. I removed this line and all tests pass and things seem to work, so I think it should be good. Change made :)

File: app/services/ajax.js [hunk 3]
```diff
@@ -1,110 +1,93 @@
...
-  }
+const defaultOptions = {
+  dataType: 'json',
+  endpoint: config.apiEndpoint || '',
+  headers: {
+    'Accept': 'application/json; version=2'
+  },
 };
 
+const PERMITTED_NON_AUTH_REQUESTS = {};
```
[the remark this answers is not in the document]
Comment (sorta) [reply app/services/ajax.js]: That makes sense to me :) Change made

File: app/services/ajax.js [hunk 4]
```diff
@@ -113,78 +96,46 @@ export default Service.extend({
...
-          return options.success.call(options.context, data, xhr.status, xhr);
+      fetch(url, fetchOptions).then(response => {
+        const { 'content-type': resContentType = '' } = response.headers.map;
+        let res;
+        if (resContentType.includes('application/json')) {
+          res = response.json();
         } else {
-          reject(xhr);
-          return options.error.call(options.context, data, xhr.status, xhr);
+          res = response.text();
```
[the remark this answers is not in the document]
Comment (sorta) [reply app/services/ajax.js]: Yeah, it's kind of funny but we do have some 2xx plaintext responses to some operations, such as cancelling a build. I don't think `travis-web` actually does anything with the content of those responses, but calling `.json` on those responses does throw an error.

[the remark this answers is not in the document]
Comment (artursmirnov) [reply app/services/ajax.js]: Ah, I see :) Thanks for the details, totally makes sense 👍

File: app/services/ajax.js [hunk 5]
```diff
@@ -113,78 +96,46 @@ export default Service.extend({
...
-          }
-        })();
-        if (xhr.status >= 200 && xhr.status < 300) {
-          resolve(data);
-          return options.success.call(options.context, data, xhr.status, xhr);
+      fetch(url, fetchOptions).then(response => {
+        const { 'content-type': resContentType = '' } = response.headers.map;
+        let res;
+        if (resContentType.includes('application/json')) {
+          res = response.json();
```
[the remark this answers is not in the document]
Comment (sorta) [reply app/services/ajax.js]: Ahh yes, good point, I think you're right. I've rearranged the function a bit :)

File: app/services/ajax.js [hunk 6]
```diff
@@ -1,110 +1,93 @@
...
+    return !authUnnecessary;
   },
 
-  patch(url, data, callback) {
-    return this.ajax(url, 'patch', {
-      data,
-      success: callback
-    });
-  },
+  request(requestUrl, mthd, opts) {
```
[the remark this answers is not in the document]
Comment (sorta) [reply app/services/ajax.js]: Good idea! I've split it up into a few different functions :)

**Comment by sorta** [comment]
Ok I think I've addressed all current feedback, so please review once you get a chance and let me know what you think. :) Thanks!

**Comment by sorta** [comment]
Excellent, thanks @artursmirnov & @piotr-travisci for the reviews!

**Comment by artursmirnov** [comment]
Hey @sorta, this PR has been [reverted](https://github.com/travis-ci/travis-web/pull/2237), because it caused an error that's described here: https://travisci.slack.com/archives/C0E7ZS68Y/p1574082016109500 - could you please take a look when you have a chance and re-submit the PR? Thanks!

**Pull request merged**